pkgs/applications/video/davinci-resolve/default.nix

```diff
@@ -1,179 +1,37 @@
-          -H 'Authority: www.blackmagicdesign.com' \
-          -H 'Cookie: _ga=GA1.2.1849503966.1518103294; _gid=GA1.2.953840595.1518103294' \
-          --data-ascii "$REQJSON" \
-          --compressed \
-          "$SITEURL")
-
-        curl \
-          --retry 3 --retry-delay 3 \
-          -H "Host: sw.blackmagicdesign.com" \
-          -H "Upgrade-Insecure-Requests: 1" \
-          -H "$USERAGENT" \
-          -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8" \
-          -H "Accept-Language: en-US,en;q=0.9" \
-          --compressed \
-          "$RESOLVEURL" \
-          > $out
-      '';
-
-      # The unpack phase won't generate a directory
-      setSourceRoot = ''
-        sourceRoot=$PWD
-      '';
-
-      installPhase = ''
-        runHook preInstall
-
-        export HOME=$PWD/home
-        mkdir -p $HOME
-
-        mkdir -p $out
-        appimage-run ./DaVinci_Resolve_${version}_Linux.run -i -y -n -C $out
-
-        mkdir -p $out/{configs,DolbyVision,easyDCP,Fairlight,GPUCache,logs,Media,"Resolve Disk Database",.crashreport,.license,.LUT}
-        runHook postInstall
-      '';
-
-      dontStrip = true;
-
-      postFixup = ''
-        for program in $out/bin/*; do
-          isELF "$program" || continue
-          addOpenGLRunpath "$program"
-        done
-
-        for program in $out/libs/*; do
-          isELF "$program" || continue
-          if [[ "$program" != *"libcudnn_cnn_infer"* ]];then
-            echo $program
-            addOpenGLRunpath "$program"
-          fi
-        done
-      '';
-    }
-  );
+  generic = args:
+    let
+      imported = import ./generic.nix args;
+    in
+    callPackage imported { };
 in
-buildFHSUserEnv {
-  name = "davinci-resolve";
-  targetPkgs = pkgs: with pkgs; [
-    librsvg
-    libGLU
-    libGL
-    xorg.libICE
-    xorg.libSM
-    xorg.libXxf86vm
-    xorg.libxcb
-    udev
-    opencl-headers
-    alsa-lib
-    xorg.libX11
-    xorg.libXext
-    expat
-    zlib
-    libuuid
-    bzip2
-    libtool
-    ocl-icd
-    glib
-    libarchive
-    xdg-utils # xdg-open needed to open URLs
-    python
-    # currently they want python 3.6 which is EOL
-    #python3
-  ];
+rec {
+
+  stable = generic {
+    version = "17.4.6";
+    pname = "davinci-resolve";
+    downloadid = "0e34f66357634628b918e96b6680c137";
+    referid = "95925530c5fd4572a47debdcdc2e3f23";
+    outputHash = "0cxhvn2i6j0ji2wk3qwq3gc70vx3vfvh5hn81l7km9n9kfxbc0nd";
+  };
 
-  runScript = "${bash}/bin/bash ${
-    writeText "davinci-wrapper"
-    ''
-    export QT_XKB_CONFIG_ROOT="${xkeyboard_config}/share/X11/xkb"
-    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${davinci}/libs
-    ${davinci}/bin/resolve
-    ''
-  }";
+  beta = generic {
+    version = "17.0b10";
+    pname = "davinci-resolve-beta";
+    downloadid = "12e3370b90704a3eb632574b67f51425";
+    referid = "73eca88216ee485a86a44dfd53692c55";
+    # I cant get past the download error.. still troubleshooting
```

Comment:
Keep it up! Im sure youll figure out whats going on

Reply:
As I noted above -- I don't think it is possible to get past that download error.  As far as I can tell there is currently *no* beta version of Davinci Resolve available for download. The 17.0b10 beta was released in Feb 2021 and is no longer available for download now that Resolve 17.1 has been released.

As far as I can tell, there are no betas for Resolve 18 available yet.

Reply:
In an attempt to make me look wrong, black magic design released the Davinci Resolve 18 beta today. So there is now a beta version available. I am looking forward to trying it since 17 does nothing but crash on startup for me.

Reply:
isnt that always how it works out :laughing: